lib/auth/oidc.go

```diff
@@ -125,6 +126,18 @@ func oidcConfig(conn types.OIDCConnector) oidc.ClientConfig {
 		// open id notifies provider that we are using OIDC scopes
 		Scope: apiutils.Deduplicate(append([]string{"openid", "email"}, conn.GetScope()...)),
 	}
+
+	if insecure {
+		conf.HTTPClient = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
```

Comment:
How is this change related to the issue? Is this just for testing?

If yes, could we change the tests so it's always TLS and revert this part?

Reply:
Yes, this is for testing. However, `TestUserInfoBlockHTTP` specifically tests a non-TLS server.

Reply:
Is that a feasible real-world scenario? I'm wary of adding code that has no production value, especially code that allows a relaxation of security parameters.

Reply:
You're right, there shouldn't be a non-testing scenario where we use this; I've moved this to tests only.

Reply:
Much better, thanks.